roles/openshift_master/tasks/systemd_units.yml

```diff
@@ -38,6 +38,11 @@
   when: create_ha_unit_files | changed
 # end workaround for missing systemd unit files
 
+- name: Preserve Master API Proxy Config options
+  command: grep PROXY /etc/sysconfig/{{ openshift.common.service_type }}-master-api
+  register: master_api_proxy
+  when: openshift.master.ha is defined and openshift.master.ha | bool and openshift_master_cluster_method == "native"
```

Comment:
So, one of the things that I keep coming back to lately is the thought that openshift_facts should really just return the state of the system as configured rather than some supposed "desired state".

My original thought was that we needed to persist these settings on the hosts somewhere in the event that someone ran the installer (or playbooks) without having the proper variables set.  Instead, I think the only time we should be worrying about "desired config" rather than current config is when someone is managing their environment using ansible, in which case the ansible variables will already be set.

In this case, openshift-facts should be returning the data on disk for these values, and we should feed in something like this to the templates (pseudocode ahead):
```
MY_PROXY_VAR={{ openshift_my_proxy_var | default(openshift.common.my_proxy_var | default(None)) }}"
```